src/Lecture1.hs

```diff
@@ -108,7 +128,14 @@ and finds a sum of the numbers inside this string.
 
 The string contains only spaces and/or numbers.
 -}
-strSum str = error "TODO"
+strSum :: String -> Int
+strSum str = 
+  sum (map convertStrToInt (words str))
+  where 
+    convertStrToInt :: String -> Int 
+    convertStrToInt x =
+        read x :: Int
```

Comment:
As you specify the top-level type signature for this function, the compiler would be able to figure out what type `read` function has in this case due to type inference, so you can skip specifying it in the body as `:: Int`. After that, you can remove the helper function `convertStrToInt` and get a nice one-liner in the end:

```suggestion
strSum str = sum (map read (words str))
```